### PR TITLE
Make node versions dynamic

### PR DESCRIPTION
**Pull request by gurgunday**
This will let us use the centralized workflow for most _libraries_

Examples of where it will be used if merged:

https://github.com/fastify/proxy-addr/pull/61#pullrequestreview-2003940510
https://github.com/fastify/fluent-json-schema/pull/242
https://github.com/fastify/process-warning/pull/104

and other `@fastify/libraries` repos

### CODE REVIEW

File: .github/workflows/plugins-ci-postgres.yml [hunk 1]
```diff
@@ -98,8 +103,8 @@ jobs:
       contents: read
     strategy:
       matrix:
-        node-version: [18, 20, 21]
-        os: [ubuntu-latest]
+        node-version: ${{ fromJson(inputs.node-versions) }}
+        os: [macos-latest, ubuntu-latest, windows-latest]
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci-postgres.yml:107]: The workflows that were only using Ubuntu should stay using only Ubuntu, as you cannot use MacOS or Windows with services.

See https://docs.github.com/en/actions/using-containerized-services/about-service-containers#about-service-containers

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: .github/workflows/plugins-ci-postgres.yml (hunk 1 above)
Comment (gurgunday) [reply .github/workflows/plugins-ci-postgres.yml:107]: Oh yes, sorry for that

Will fix

**Comment by gurgunday** [comment]
Should package managers be tested cross platform?

**Comment by Fdawgs** [comment]
> Should package managers be tested cross platform?

Probably leave those on Ubuntu only as well, so they match what the main Fastify repo is doing. :)

If you could update the readme with the new option as well then this PR'll be good to go!

**Comment by gurgunday** [comment]
Done, let me know if the README could be improved

**Review by Fdawgs — APPROVED** [review]

**Comment by gurgunday** [comment]
Just reverted an unnecessary change to benchmark

815aca9f00ba4da2502550b5adb1afc18752fc5b

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**